src/distilabel/pipeline.py

```diff
@@ -384,16 +436,28 @@ def _build_dataset(  # noqa: C901
         if self.labeller is None:
             labels = [{} for _ in range(len(dataset))]
         else:
-            # If the LLM returns futures, we need to wait for them to finish
             processed_labels = []
-            if self.labeller.return_futures:
-                for future in batch_labels:
+            if self.labeller.return_futures is not None:
+                for i, future in enumerate(labels, 1):
```

Comment:
For readability I feel more comfortable with kwargs

```suggestion
                for i, future in enumerate(labels, start=1):
```